packages/gen-ai-hub/src/utils/deployment-cache.ts

```diff
@@ -0,0 +1,83 @@
+// TODO: should we move this to @sap-cloud-sdk/connectivity? or add some kind of exception to the rule if possible?
+// eslint-disable-next-line import/no-internal-modules
+import { Cache } from '@sap-cloud-sdk/connectivity/internal.js';
+import { type AiDeployment } from '@sap-ai-sdk/ai-core';
+import { type DeploymentResolutionOptions } from './deployment-resolver.js';
+import { extractModel, type FoundationModel } from './model.js';
+
+function getCacheKey({
+  scenarioId,
+  executableId,
+  model,
+  resourceGroup = 'default'
```

Comment:
If we set the default for the resource group here, we might as well do the same for all the other parameters?
At least this would be consistent and makes the string interpolation less loaded.

Reply:
I can only do this for `executableId`, that is a good idea.